Feature/1 migrate collecting logic and repo setup from origin repo (#2)
AbsaOSS/living-doc-collector-gh: doc-issues/README.md

  - **Example**:
    ```yaml
    with:
      doc-issues: true
    ```
    
## GitHub Repository Structure Requirements

- Is recommended to use only one dedicated repository for documentation tickets (GitHub issues with supported labels).
- User Story **can** point to multiple Features
- Feature **can** point to multiple Functionalities
- Functionality **should** point to one Feature
  - Why one only?
    - To avoid confusion and ensure clarity in the documentation.
    - Functionality implements a specific aspect of a feature, and linking it to multiple features can create ambiguity.

---

## Usage

See the default minimal Living Documentation Issues mode action step definition:

```yaml
- name: Living Documentation Collect for Github
  id: living_doc_collector_gh
  uses: AbsaOSS/living-doc-collector-gh@v0.1.0
  env:
    GITHUB-TOKEN: ${{ secrets.REPOSITORIES_ACCESS_TOKEN }}  
  with:
    doc-issues: true                   # living documentation issues mode de/activation  
    doc-issues-repositories: |
        [
          {
            "organization-name": "fin-services",
            "repository-name": "investment-app"
          }
        ]
```

> See the full example of Living Documentation Issues mode step definition (in example are used non-default values):

Reviewer: ```suggestion
See the full example of the Living Documentation Issues mode step definition (in the example non-default values are used):
```

- **Example**

  ```yaml
  with:
    doc-issues-repositories: |
      [
        {
          "organization-name": "your-organization-name",
          "repository-name": "your-project-living-documentation",
        },
        {
          "organization-name": "your-organization-name",
          "repository-name": "your-another-project-living-documentation",
          "projects-title-filter": ["Management Overview"]
        }
      ]
  
      doc-issues-project-state-mining: true 
  ```

---
## Expected Output

> The Living Documentation Collector for GitHub is designed to produce a collection of Consolidated Issues. Where **consolidated** means that the issues information is merged from the Repository Issues and from the Project Issues. 

Reviewer: ```suggestion
The Living Documentation Collector for GitHub is designed to produce a collection of Consolidated Issues. Where **consolidated** means that the issues' information is merged from both the Repository Issues and from the Project Issues. 
```

The mode produces the file `output/collected_gh_consolidated_issues.json` with the following structure:

```
TODO 
```

The `output` folder is the root output directory for the action.

---
## Documentation Ticket Introduction

> A **Documentation Ticket** is a small piece of documentation realized as a GitHub Issue dedicated to project documentation. Unlike development-focused tickets, Documentation Ticket can remain in open state continuously, evolving as updates are needed, and can be reopened or revised indefinitely. They are not directly tied to Pull Requests (PRs) but can be referenced for context.

Reviewer: ```suggestion
A **Documentation Ticket** is a small piece of documentation realized as a GitHub Issue dedicated to project documentation. Unlike development-focused tickets, a Documentation Ticket can remain in open state continuously, evolving as updates are needed, and can be reopened or revised indefinitely. They are not directly tied to Pull Requests (PRs) but can be referenced for context.
```

### Hosting Documentation Tickets in a Solo Repository

Using a dedicated repository solely for documentation tickets provides multiple advantages:
- **Streamlined Management:** This avoids cross-project conflicts, board exclusions and enables specialized templates solely for documentation purposes.- **Focused Access Control:** This allows a small team to manage and edit documentation without interference, maintaining high-quality content.
- **Optimized Data Mining:** Supports easier and more efficient data extraction for feedback and review cycles through Release Notes.
- **Implementation Reflection:** Mirrors elements from the implementation repositories, providing a high-level knowledge source that is valuable for both business and technical teams.
- **Release Notes Integration:** Documentation can evolve based on insights from release notes, serving as a dynamic feedback loop back to the documentation repository.

---

## Living Documentation Issues Mode Features

### Issues Data Mining from GitHub Repositories

> This is a build-in feature, that allows you to define which repositories should be included in the living documentation issues mode process. This essential process can not be deactivated inside of mode scope. By specifying repositories, you can focus on the most relevant projects for your documentation needs.

Reviewer: ```suggestion
This is a built-in feature, that allows you to define which repositories should be included in the living documentation issues mode process. This essential process cannot be deactivated inside of mode scope. By specifying repositories, you can focus on the most relevant projects for your documentation needs.
```
